core/src/main/java/com/linecorp/armeria/client/DefaultWebClient.java

```diff
@@ -106,8 +106,10 @@ private HttpResponse execute(EndpointGroup endpointGroup, HttpRequest req, Sessi
             final IllegalArgumentException cause = new IllegalArgumentException("invalid path: " + req.path());
             return abortRequestAndReturnFailureResponse(req, cause);
         }
+        final String newPath = pathWithQuery(pathAndQuery.path(), pathAndQuery.query());
+        final HttpRequest newReq = req.withHeaders(req.headers().toBuilder().path(newPath));
```

Comment:
Could you check whether `newPath` is different from `req.path()` before mutating `req.headers()`